ksqldb-functional-tests/src/test/resources/query-validation-tests/joins.json

```diff
@@ -3234,18 +3210,6 @@
         "type": "io.confluent.ksql.util.KsqlStatementException",
         "message": "Implicit repartitioning of windowed sources is not supported. See https://github.com/confluentinc/ksql/issues/4385."
       }
-    },
-    {
-      "name": "stream-table key-to-key with necessary repartition - SR-enabled key format",
```

Comment:
This test also seems redundant to the two from above?

Reply:
This test is not redundant from the other two. We have special logic that permits repartitioning tables for joins on sources with SR-enabled key formats (in order to ensure copartitioning in light of potentially different schema IDs). This test ensures that this feature does not allow changing the primary key column.

Reply:
If I understand you comment correctly, the test does not test what it is supposed to test? With this PR, we would fail in `verifyJoin` already as we try to join on a non-key table attribute (`T.VAL`) that is always disallowed.

Reply:
With the changes in this PR, yeah. I'm fine removing this test from here. It'd be nice to update the new test for this case (that we only allow joins on the primary key of the table) to use an SR-enabled key format as an extra safe-guard.